railgun/internal/ctrlutils/utils.go

```diff
@@ -141,3 +146,31 @@ func CRDExists(client client.Client, gvr schema.GroupVersionResource) bool {
 	}
 	return true
 }
+
+// retrieve Kong Admin API URL from configured name/namespace service
+func RetrieveKongAdminAPIURL(ctx context.Context, KongAdminAPI string, kubeCfg *rest.Config) (string, error) {
+	namespace, name, err := util.ParseNameNS(KongAdminAPI)
+	if err != nil {
+		return "", fmt.Errorf("failed to pars kong admin api namespace and name. err %v", err)
+	}
+	CoreClient, _ := clientset.NewForConfig(kubeCfg)
+	svc, err := CoreClient.CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to creating kubernetes client")
+	}
+	ingresses := svc.Status.LoadBalancer.Ingress
+	adminIP := ""
+	for _, ingress := range ingresses {
+		adminIP = ingress.IP
+	}
```

Comment:
This implementation is bound to work most if not all the time, however it has two technical flaws:

1. it can technically _leave the `adminIP` as an empty string `""`_
2. it doesn't account for the technical possibility of multiple `Ingress` entries

My recommendation would be to validate that the list is exactly 1 in length, and throw an error if not.

Reply:
A loadbalancer type can not have multiple external ip/host, otherwise it is against ip uniqueness.

Reply:
setting aside multiple IPs, it could still be an empty list right? So that could be a problem we could do a short validation check for?